.github/workflows/canopeum_backend.yml

```diff
@@ -28,13 +30,21 @@ jobs:
         working-directory: canopeum_backend
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+      - name: Install uv using the standalone installer
+        run: irm https://astral.sh/uv/install.ps1 | iex
+        shell: powershell
+      - name: Restore uv cache
+        uses: actions/cache@v4
         with:
-          python-version: "3.12"
-          cache: pip
-          cache-dependency-path: "**/requirements*.txt"
-      - run: pip install -r requirements-dev.txt
+          path: ${{ env.UV_CACHE_DIR }}
+          key: uv-${{ runner.os }}-${{ hashFiles('uv.lock') }}
+          restore-keys: |
+            uv-${{ runner.os }}-${{ hashFiles('uv.lock') }}
+            uv-${{ runner.os }}
+      - run: uv sync --locked --extra dev
       - run: mypy . --python-version=3.12
+      - name: Minimize uv cache
+        run: uv cache prune --ci
```

Comment:
After consideration, I'll avoid dependency caching for now. The speedup in install from having to redownload is mostly offset by the caching overhead. Avoiding potentially random network failure is nice, but rare. And I'd prefer other maintainers not having to handle maintaining this piece of devops.
https://docs.astral.sh/uv/guides/integration/github/#caching